src/test/java/se/sundsvall/precheck/service/PreCheckServiceTest.java

```diff
@@ -0,0 +1,120 @@
+    @InjectMocks
+    private PreCheckService preCheckService;
+
+    @Test
+    void testCheckPermitWhen2xxStatusCodeButNoBody() {
+        // Arrange
+
+
+        Mockito.when(citizenClient.getCitizen(Mockito.eq(partyId)))
+                .thenReturn(ResponseEntity.ok().build());
+
+        Mockito.when(partyAssetsClient.getPartyAssets(Mockito.eq(partyId), Mockito.anyString()))
+                .thenReturn(ResponseEntity.ok().build());
+
+        // Act
+        try {
+            preCheckService.checkPermit(partyId, municipalityId, assetType);
+        } catch (Exception e) {
+            // Assert
+            assertEquals(
+                    Problem.valueOf(FORBIDDEN, String.format(INVALID_MUNICIPALITY_ID, partyId)).getMessage(),
+                    e.getMessage()
+            );
+            assertEquals(
+                    Problem.valueOf(FORBIDDEN).getStatus(),
+                    ((Problem) e).getStatus()
+            );
+        }
+
+        verify(citizenClient, times(1)).getCitizen(partyId);
+        verify(partyAssetsClient, times(1)).getPartyAssets(partyId, String.valueOf(Status.ACTIVE));
+    }
+
+    @Test
+    void testIntegrationErrors() {
+        testErrorCase(ResponseEntity.noContent().build(), ResponseEntity.ok().build());
+        testErrorCase(ResponseEntity.ok().build(), ResponseEntity.noContent().build());
+        testErrorCase(ResponseEntity.status(FORBIDDEN.getStatusCode()).build(), ResponseEntity.ok().build());
+        testErrorCase(ResponseEntity.ok().build(), ResponseEntity.status(FORBIDDEN.getStatusCode()).build());
+
+        verify(citizenClient, times(4)).getCitizen(partyId);
+        verify(partyAssetsClient, times(4)).getPartyAssets(partyId, String.valueOf(Status.ACTIVE));
+    }
+
+    private void testErrorCase(ResponseEntity<CitizenExtended> citizenResponse, ResponseEntity<List<Asset>> assetsResponse) {
+        Mockito.when(citizenClient.getCitizen(Mockito.eq(partyId))).thenReturn(citizenResponse);
+        Mockito.when(partyAssetsClient.getPartyAssets(Mockito.eq(partyId), Mockito.anyString())).thenReturn(assetsResponse);
+
+        try {
+            preCheckService.checkPermit(partyId, municipalityId, assetType);
+        } catch (Exception e) {
+            assertEquals(Problem.valueOf(NO_CONTENT, String.format(NO_CONTENT_FOUND, partyId)).getMessage(),
+                    e.getMessage());
+            assertEquals(Problem.valueOf(NO_CONTENT).getStatus(), ((Problem) e).getStatus());
+        }
+    }
+
+    @Test
+    void testCheckPermitWhenResourceNotFound() {
+        Mockito.when(citizenClient.getCitizen(Mockito.eq(partyId)))
+                .thenReturn(ResponseEntity.noContent().build());
+
+        Mockito.when(partyAssetsClient.getPartyAssets(Mockito.eq(partyId), Mockito.anyString()))
+                .thenReturn(ResponseEntity.noContent().build());
+
+        // Act
+        try {
+            preCheckService.checkPermit(partyId, municipalityId, assetType);
```

Comment:
assertThrows is usually used when asserting thrown exceptions.

Reply:
Yes, it is more proper to use **assertThrows** here. 
I will make the change right away.